Mathlib/Condensed/Explicit.lean

```diff
@@ -0,0 +1,160 @@
+  rw [isSheaf_coherent_iff_regular_and_extensive]
+  apply and_congr
+  · exact isSheaf_iff_preservesFiniteProducts F
+  · exact isSheaf_iff_equalizerCondition F
+
+theorem isSheaf_iff_preservesFiniteProducts_and_equalizerCondition'
+    {A : Type (u+2)} [Category.{u+1} A] (G : A ⥤ Type (u+1))
+    [HasLimits A] [PreservesLimits G] [ReflectsIsomorphisms G] (F : CompHaus.{u}ᵒᵖ ⥤ A) :
+    Presheaf.IsSheaf (coherentTopology CompHaus) F ↔
+    Nonempty (PreservesFiniteProducts (F ⋙ G)) ∧ EqualizerCondition (F ⋙ G) := by
+  rw [Presheaf.isSheaf_iff_isSheaf_forget (coherentTopology CompHaus) F G,
+    isSheaf_iff_isSheaf_of_type, isSheaf_iff_preservesFiniteProducts_and_equalizerCondition]
+
+instance {A B : Type*} [Category A] [Category B] (F : B ⥤ A) (E : A)  [PreservesFiniteProducts F] :
+    PreservesFiniteProducts (F ⋙ coyoneda.obj (op E)) :=
+  ⟨fun J _ ↦ @compPreservesLimitsOfShape _ _ _ _ _ _ _ _ F (coyoneda.obj (op E))
+    (PreservesFiniteProducts.preserves J) ((preservesLimitsOfSizeShrink _).preservesLimitsOfShape)⟩
+
+end CompHaus
+
+namespace Profinite
+
+theorem isSheaf_iff_preservesFiniteProducts_and_equalizerCondition
+    (F : Profinite.{u}ᵒᵖ ⥤ Type (u+1)) :
+    IsSheaf (coherentTopology Profinite) F ↔
+    Nonempty (PreservesFiniteProducts F) ∧ EqualizerCondition F := by
+  rw [isSheaf_coherent_iff_regular_and_extensive]
+  apply and_congr
+  · exact isSheaf_iff_preservesFiniteProducts F
+  · exact isSheaf_iff_equalizerCondition F
+
+theorem isSheaf_iff_preservesFiniteProducts_and_equalizerCondition'
+    {A : Type (u+2)} [Category.{u+1} A] (G : A ⥤ Type (u+1))
+    [HasLimits A] [PreservesLimits G] [ReflectsIsomorphisms G] (F : Profinite.{u}ᵒᵖ ⥤ A) :
+    Presheaf.IsSheaf (coherentTopology Profinite) F ↔
+    Nonempty (PreservesFiniteProducts (F ⋙ G)) ∧ EqualizerCondition (F ⋙ G) := by
+  rw [Presheaf.isSheaf_iff_isSheaf_forget (coherentTopology Profinite) F G,
+    isSheaf_iff_isSheaf_of_type, isSheaf_iff_preservesFiniteProducts_and_equalizerCondition]
+
+end Profinite
+
+namespace Stonean
+
+theorem isSheaf_iff_preservesFiniteProducts
+    (F : Stonean.{u}ᵒᵖ ⥤ Type (u+1)) :
+    IsSheaf (coherentTopology Stonean) F ↔ Nonempty (PreservesFiniteProducts F) := by
+  rw [isSheaf_coherent_iff_regular_and_extensive, and_iff_left ?_]
+  · exact CategoryTheory.isSheaf_iff_preservesFiniteProducts F
+  · rw [Presieve.isSheaf_coverage]
+    intro X R ⟨Y, hR⟩
+    have _ : R.regular := ⟨Y, hR⟩
+    exact isSheafFor_regular_of_projective R F
+
+theorem isSheaf_iff_preservesFiniteProducts'
+    {A : Type (u+2)} [Category.{u+1} A] (G : A ⥤ Type (u+1))
+    [HasLimits A] [PreservesLimits G] [ReflectsIsomorphisms G] (F : Stonean.{u}ᵒᵖ ⥤ A) :
+    Presheaf.IsSheaf (coherentTopology Stonean) F ↔
+    Nonempty (PreservesFiniteProducts (F ⋙ G)) := by
+  rw [Presheaf.isSheaf_iff_isSheaf_forget (coherentTopology Stonean) F G,
+    isSheaf_iff_isSheaf_of_type, isSheaf_iff_preservesFiniteProducts]
+
+end Stonean
+
+namespace Condensed
```

Comment:
Should this maybe be `CondensedSet`? Because the results here are not about arbitrary `Condensed FooBar`, but specifically about `CondensedSet`. That would allow more use of dot-notation...
Has this been discussed before and/or is there some policy? I'm not aware of it...

Reply:
The target category should actually be more general. Then I can add versions for `Type (u+1)` in the `CondensedSet` namespace (maybe as `abbrev`s?)